plugins/common/socket/datagram.go

```diff
@@ -98,26 +100,32 @@ func (l *packetListener) listenConnection(onConnection CallbackConnection, onErr
 				break
 			}
 
-			// Decode the contents depending on the given encoding
-			body, err := l.decoder.Decode(buf[:n])
-			if err != nil && onError != nil {
-				onError(fmt.Errorf("unable to decode incoming packet: %w", err))
-			}
+			d := make([]byte, n)
+			copy(d, buf[:n])
```

Comment:
Hmmm I wonder if we really need to copy that thing?

Reply:
We do need to copy it. Because the parsing is now done asynchronously, the buffer may already have been filled with new data when parsing starts. This results in missed data or, in some cases, mixed data where parts of one message are mixed with parts of the next message.